Client/PaymentRequest/PaymentRequestExtension.swift

```diff
@@ -0,0 +1,116 @@
+
+import Data
+import Shared
+import WebKit
+
+private let log = Logger.browserLogger
+
+let popup = PaymentHandlerPopupView(imageView: nil, title: Strings.paymentRequestTitle, message: "")
+
+class PaymentRequestExtension: NSObject {
+    fileprivate weak var tab: Tab?
+    fileprivate var response = ""
+    
+    init(tab: Tab) {
+        self.tab = tab
+    }
+}
+
+extension PaymentRequestExtension: TabContentScript {
+    static func name() -> String {
+        return "PaymentRequest"
+    }
+    
+    func scriptMessageHandlerName() -> String? {
+        return "PaymentRequest"
+    }
+    
+    func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        if message.name == "PaymentRequest", let body = message.body as? NSDictionary {
+            guard let name = body["name"] as? String, let supportedInstruments = body["supportedInstruments"] as? String, let details = body["details"] as? String else {
+                return
+            }
+            if name == "payment-request-show" {
+                do {
+                    popup.clearDisplayItems()
+                    popup.removeAllButtons()
+                    popup.addTotalLabel(message: "")
+                    
+                    guard let detailsData = details.data(using: String.Encoding.utf8), let supportedInstrumentsData = supportedInstruments.data(using: String.Encoding.utf8) else {
+                        log.error("Error parsing data")
+                        return
+                    }
+                    let details = try JSONDecoder().decode(PaymentRequestDetailsHandler.self, from: detailsData)
+                    
+                    let supportedInstruments =  try JSONDecoder().decode([PaymentRequestSupportedInstrumentsHandler].self, from: supportedInstrumentsData)
+                    
+                    for item in details.displayItems {
+                        popup.addDisplayItemLabel(message: item.label + ":  " + item.amount.value + " " + item.amount.currency + "\n")
+                        log.info(item.label)
+                    }
+                    
+                    popup.addTotalLabel(message: details.total.label + ":  " + details.total.amount.value + " " + details.total.amount.currency + "\n")
+                    
+                    popup.addButton(title: Strings.paymentRequestPay) { [weak self] in
+                        guard let self = self else {
+                            return .flyDown
+                        }
+                        guard let rewards = self.tab?.rewards, let publisher = self.tab?.publisher, let amount = Double(details.total.amount.value) else {
+                            return .flyDown
+                        }
+                        
+                        rewards.ledger.tipPublisherDirectly(publisher, amount: amount, currency: "BAT") { _ in
+                          // TODO: Handle started tip process
```

Comment:
Is there something filed for this TODO?

Reply:
https://github.com/brave/brave-ios/issues/2383 - We are using the tipping API for the demo only. I'll update the code to say that. The SKU APIs are in progress.